### PR TITLE
Admin, Enterprise Settings: Remove jQuery dirty form plugin

### PR DESCRIPTION
**Pull request by binarygit**
#### What? Why?

Closes # 

 This PR introduces an stimulusJs implementation of a `dirty form` plugin.
 
 These are the "features" that the stimulusJs implementation has: 
 
 1. Prevent users from leaving the page when the form is dirty i.e an input's value has changed.
 2. Provides an interface to enable/disable the `submit` button after determining whether the form is dirty or not.
 3. Provides an API to mark the form as dirty.

#### What should we test?

- Visit ... page.
- 

#### Release notes

Changelog Category: User facing changes | Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/webpacker/controllers/dirty_form_controller.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+const LEAVING_PAGE_MESSAGE = "Are you sure you want to leave?";
+
+export default class extends ApplicationController {
+  static values = { dirty: Boolean };
+  static targets = ["submitBtn"];
+
+  connect() {
+    // find all inputs within the form
+    // and attach eventListeners
+    this.inputs = this.element.querySelectorAll("input");
```
Comment (jibees) [inline app/webpacker/controllers/dirty_form_controller.js:11]: I think I'd use `targets`

Comment (binarygit) [reply app/webpacker/controllers/dirty_form_controller.js:11]: Good catch on the `textarea` and `select`! This is why I wanted your input on this.

I think targets are a bit too cumbersome,( we have a large form in this case and might have other large forms ), but I can definitely see that we might require to `target`  some obscure custom element or something in the future, without bloating our list of elements that need eventListeners.

Let me experiment, and try to correct the errors that are present right now, see what I'll find, and make a decision on the `targets` then.

File: app/webpacker/controllers/dirty_form_controller.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    this.textAreas = this.element.querySelectorAll("textarea");
+    this.selects = this.element.querySelectorAll("select");
+    this.elArr = [...this.inputs, ...this.textAreas, ...this.selects];
+
+    this.elArr.forEach((el) => {
+      let boundFunc = this.makeDirty.bind(this);
+      el.addEventListener("change", boundFunc);
+    });
```
Comment (mansakondo) [inline app/webpacker/controllers/dirty_form_controller.js:43]:
```suggestion
  trackChanges() {
    // find all inputs within the form
    // and attach eventListeners
    TRACKED_TAGS.forEach((tag) => {
      let targets = document.getElementsByTagName(tag)
      this.trackedTargets.push(...targets)
    })

    this.trackedTargets.forEach((target) => {
      target.addEventListener("change", this.makeDirty.bind(this))
    })
```

**Review by mansakondo — COMMENTED** [review]
I finally had some time to look at your PR. Though it's a priority at the moment, I refactored the code a little bit while I was reading it. So, I left some suggestions.

File: app/helpers/admin/enterprises_helper.rb [hunk 3]
```diff
@@ -39,7 +39,6 @@ def enterprise_side_menu_items(enterprise)
         { name: 'inventory_settings', icon_class: "icon-list-ol", show: is_shop },
         { name: 'tag_rules', icon_class: "icon-random", show: is_shop },
         { name: 'shop_preferences', icon_class: "icon-shopping-cart", show: is_shop },
-        { name: 'users', icon_class: "icon-user", show: true }
```
Comment (mansakondo) [inline app/helpers/admin/enterprises_helper.rb:42]: Was this line removed on purpose? Because the "Users" tab doesn't show up anymore.

Comment (binarygit) [reply app/helpers/admin/enterprises_helper.rb:42]: no this was intentional, there's some angular there that's messing with the page URL. Will restore it once I merge this[remove angular from users](https://github.com/openfoodfoundation/openfoodnetwork/pull/9729) into here.

Thank You for the code suggestions.

Comment (mansakondo) [reply app/helpers/admin/enterprises_helper.rb:42]: Ah, I see. Thanks for sharing your PR. Is there another PR's related to this one?

Comment (binarygit) [reply app/helpers/admin/enterprises_helper.rb:42]: There's this one https://github.com/openfoodfoundation/openfoodnetwork/pull/9761 but I just merged it into this PR itself in my latest commit.

**Comment by jibees** [comment]
I think this one is handled by another PR (can't find it) so I'll close it for now. Feel free to reopen if needed!

**Pull request closed without merging**